src/utils/components/NetworkInterfaceModal/components/NetworkInterfacePasst.tsx

```diff
@@ -1,40 +1,86 @@
-import React, { FC } from 'react';
+import React, { FC, useCallback, useMemo, useState } from 'react';
 import { Link } from 'react-router-dom-v5-compat';
 
+import SelectToggle from '@kubevirt-utils/components/toggles/SelectToggle';
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
 import useNamespaceUDN from '@kubevirt-utils/resources/udn/hooks/useNamespaceUDN';
+import { interfaceTypesProxy } from '@kubevirt-utils/resources/vm/utils/network/constants';
 import usePasstFeatureFlag from '@overview/SettingsTab/PreviewFeaturesTab/hooks/usePasstFeatureFlag';
-import { Checkbox, FormGroup, Popover, Split, SplitItem } from '@patternfly/react-core';
+import { FormGroup, Popover, Select, SelectOption, Split, SplitItem } from '@patternfly/react-core';
 import { HelpIcon } from '@patternfly/react-icons';
 
+import {
+  BINDING_SELECT_DESCRIPTION,
+  BINDING_SELECT_TITLE,
+  L2BRIDGE_BINDING,
+  PASST_BINDING,
+} from './constants';
+
 type NetworkInterfacePasstProps = {
   namespace: string;
-  onChange: (enabled: boolean) => void;
   passtEnabled: boolean;
+  setInterfaceType: (newValue: string) => void;
 };
 
 const NetworkInterfacePasst: FC<NetworkInterfacePasstProps> = ({
   namespace,
-  onChange,
   passtEnabled,
```

Comment:
we can pass the the selected interface type directly and avoid mappings between the flag and the type